### PR TITLE
[FIX] throw warning when NiftiLabelMasker resamples images at transform time

### LINKED ISSUE #6040: [BUG] Throw a warning when images and label image have different fov at transform time in NiftiLabelsMasker
#6028 solves the problem for NiftiMapsMasker, it should be symmetrically done for NiftiLabelsMasker

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #6040

Use of AI tools:
- [ ] LLM tools were used to generate at least part of the content of this pull-request.
- [x] No LLM tools were used to generate any part of the content of this pull-request.

Changes proposed in this pull request:

- throw warning when NiftiLabelMasker resamples images at transform time

### CODE REVIEW

**Review by bthirion — APPROVED** [review]

File: nilearn/_utils/estimator_checks.py [hunk 1]
```diff
@@ -3502,16 +3502,8 @@ def check_masker_transform_resampling(estimator_orig) -> None:
...
-                with pytest.warns(UserWarning, match="at transform time"):
-                    estimator.transform(imgs_with_different_fov)
-            else:
-                with warnings.catch_warnings(record=True) as warning_list:
-                    estimator.transform(imgs_with_different_fov)
-                assert all(
-                    "at transform time" not in str(x.message)
-                    for x in warning_list
-                )
+            with pytest.warns(UserWarning, match="at transform time"):
```
Comment (Remi-Gau) [inline nilearn/_utils/estimator_checks.py]:
```suggestion
            with pytest.warns(UserWarning, match="Resampling images at transform time"):
```

File: nilearn/maskers/tests/test_nifti_maps_masker.py [hunk 2]
```diff
@@ -534,28 +534,3 @@ def test_nifti_maps_masker_overlap(maps_img_fn, allow_overlap, img_fmri):
             masker.fit_transform(img_fmri)
     else:
         masker.fit_transform(img_fmri)
-
-
-def test_nifti_maps_masker_transform_resample_warning(img_fmri):
```
Comment (Remi-Gau) [inline nilearn/maskers/tests/test_nifti_maps_masker.py:539]: there is now an equivalent test in estimator_checks
